tests/tel2puml/find_unique_graphs/test_json_data_converter.py

```diff
@@ -222,3 +228,73 @@ def test_build_nested_dict() -> None:
         parsed_data4 = "string_value"
         with pytest.raises(TypeError):
             _build_nested_dict(path_segments4, parsed_data4)
+
+    @staticmethod
+    def test_extract_nested_value() -> None:
+        """Tests the function _extract_nested_value"""
+        sample_data = {
+            "resource": {
+                "attributes": [
+                    {
+                        "service": {"name": "frontend"},
+                        "response": "200",
+                        "spans": [{"trace_id": "001"}, {"span_id": "002"}],
+                    }
+                ]
+            },
+            "host": [{"provider": "receiever"}],
+        }
+
+        # Test simple case
+        path = "host::provider"
+        assert _extract_nested_value(sample_data, path) == {
+            "provider": "receiever"
+        }
+
+        # Test nested dictionary before list
+        path = "resource:attributes::response"
+        assert _extract_nested_value(sample_data, path) == {"response": "200"}
+
+        # Test nested dictionary before and after list
+        path = "resource:attributes::service:name"
+        assert _extract_nested_value(sample_data, path) == {
+            "service:name": "frontend"
+        }
+
+        # Test larger nested data structure after list
+        path = "resource:attributes::spans"
+        assert _extract_nested_value(sample_data, path) == {
+            "spans": [{"trace_id": "001"}, {"span_id": "002"}]
```

Comment:
More of a question - why are we not keeping any of the paths in the return dict that come before a list?

Reply:
The path before the list is the key within header_dict. In the function process_header we have:
```
    key = path.split("::")[0]
    if isinstance(value, (dict, list)):
        header_dict[key] = _flatten_json_dict(value)
    else:
        header_dict[key] = value
```
key is equal to the path before the list

Reply:
Ok think I understand